Infrastructure/Authentication/UserRepository.cs

```diff
@@ -0,0 +1,88 @@
+using System.Data;
+using Dapper;
+using FluentResults;
+using FluentValidation.Validators;
+using Infrastructure.Authentication.Models;
+using Infrastructure.Persistence.Contracts;
+using Microsoft.Extensions.Logging;
+
+namespace Infrastructure.Authentication;
+
+public class UserRepository
+{
+	private readonly IDbConnectionFactory _connection;
+	private readonly ILogger<UserRepository> _logger;
+	
+	public UserRepository(IDbConnectionFactory connection, ILogger<UserRepository> logger)
+	{
+		_connection = connection;
+		_logger = logger;
+	}
+
+	public async Task<User?> GetUserByIdAsync(int userId)
+	{
+		using var con = await _connection.CreateConnectionAsync();
+		var query = """
+		            SELECT * FROM users WHERE id = @id;
+		            """;
+		var user = await con.QueryAsync<User>(query, new { id = userId });
+		return user.FirstOrDefault();
+	}
+
+	public async Task<User?> GetUserDetailsByEmailAsync(string email)
+	{
+		using var con = await _connection.CreateConnectionAsync();
+		var query = """
+					SELECT users.id, users.email, users.password_hash AS passwordhash, role.name AS rolestr FROM users
+						JOIN user_role ON users.id = user_role.user_id
+						JOIN role ON user_role.role_id = role.id
+					    WHERE email = @email;
+					""";
+		var user = await con.QuerySingleAsync<User>(query, new { email });
```

Comment:
Change from single query to fit the situation where a User has more than one role